roles/podman/templates/container-.service.j2

```diff
@@ -0,0 +1,46 @@
+# vi: ft=systemd
+[Unit]
+Description=Podman container service for {{ container.name }}
+Wants=network-online.target
+After=network-online.target
+RequiresMountsFor=/var/run/container/storage
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+Type=forking
+
+PIDFile=%t/%n-pid
+Restart=on-failure
+RuntimeDirectory=%n
+
+ExecStartPre=/bin/rm -f %t/%n-pid %t/%n-cid
+# QUESTION: Why two cgroups? This was taken from:
+# https://github.com/python-discord/infra/issues/18#issuecomment-1027307623
+# TODO: make publish can be looped over to get multiple
+# ports opened.
```

Comment:
@imsofi Do you think this is resolved now that arbitrary options can be passed to podman? I imagine most services will only need to publish one port, so there isn't much benefit to the small convenience of supporting a list of ports directly.

Reply:
No, arbitary arguments does not help adding extra options to the start command im afraid.

Using arbitrary options to simplify the complexity in our role is counter productive. As we are essentially pushing that complexity upwards. Where now everyone using the podman role now also needs to know how to configure systemd, leading to a very over-complicated interface to set up podman containers.

Reply:
It's not using systemd options; it's https://github.com/python-discord/infra/pull/57/commits/3a826423014cf7c006feb05d74cf456a8abdb200

There is some merit to what you are saying, but I also think it's not unreasonable to expect some options to be known. I think it's a fair trade off for more flexibility. It seems tedious to maintain a template that needs to cover all options we need. Furthermore, because now instead of being able to leverage existing knowledge of podman/systemd, one has to study our specific template to see how to set an option.